File: 240418/두 사람/two-person.js
const fs = require('fs');

class Person {
    #age;
    #gender;

    constructor(age, gender){
        this.#age = age;
        this.#gender = gender;
    }

    get age(){
        return this.#age;
    }

    get gender(){
        return this.#gender;
    }
}

const input = fs.readFileSync(0).toString().split("\n");

const person1 = new Person(...(input[0].split(" ")));
const person2 = new Person(...(input[1].split(" ")));

if((person1.age >= 19 && person1.gender === 'M') || (person2.age >= 19 && person2.gender === "M")){
    console.log(1);
}else{
    console.log(0);
}

File: 240418/두 사람/two-person.java
import java.util.*;

public class Main {
    public static class Person {
        int age;
        String gender;

        public Person(int age, String gender){
            this.age = age;
            this.gender = gender;
        }
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        Person person1 = new Person(sc.nextInt(), sc.next());
        Person person2 = new Person(sc.nextInt(), sc.next());

        if((person1.age >= 19 && person1.gender.equals("M")) || (person2.age >= 19 && person2.gender.equals("M"))){
            System.out.println(1);
        }else{
            System.out.println(0);
        }
        
    }
}